tests/parser/features/test_assert.py

```diff
@@ -1,17 +1,17 @@
-import pytest
 
 
-def test_assert_refund(t, get_contract_with_gas_estimation):
+def test_assert_refund(w3, tester, get_contract_with_gas_estimation, assert_tx_failed):
```

Comment:
Curious of the choice to use both `w3` and `tester` here.  `w3.eth.getBalance(...)` should allow you to avoid direct `tester` interactions which is mostly just academic, but it does reduce how closely your API is tied to the `eth-tester` API (and the `w3` API is likely to be more stable).

Reply:
I agree, for tests direct _tester_ acces should be minimised, and w3 optimised will change accordingly ;)